.github/workflows/release-branch.yml

```
@@ -0,0 +1,68 @@
+name: Automation - Release Branch
+
+on:
+  workflow_dispatch:
+    inputs:
+      release-series:
+        description: 'Release series (e.g., v0.85.x). If blank (i.e. bugfix release), uses most recent release/v* branch'
+        required: false
+
+jobs:
+  release-branch:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          fetch-depth: 0
+
+      - name: Setup Go
+        uses: actions/setup-go@f111f3307d8850f501ac008e886eec1fd1932a34 # v5.3.0
+        with:
+          go-version: 1.23.9
+
+      - name: Setup Git config
+        run: |
+          git config --global user.name "opentelemetrybot"
+          git config --global user.email "107717825+opentelemetrybot@users.noreply.github.com"
+
+      - name: Find most recent release branch
+        id: find-release-branch
+        if: ${{ !inputs.release-series }}
+        run: |
+          MOST_RECENT_RELEASE=$(git branch -r | grep "origin/release/v" | sort -Vr | head -n 1 | sed 's|^[[:space:]]*origin/release/||')
```

nit: It seems like you can do this to avoid the final sed
```suggestion
          MOST_RECENT_RELEASE=$(git branch -r --format='%(refname:short)' | grep "origin/release/v" | sort -Vr | head -n 1)
```